### PR TITLE
configure crowdsec by api

### PR DESCRIPTION
**Pull request by stephdl**
- Configure crowdsec with configure-module
- destroy-module fix the traefik route removal because we do not use it
- fix the smarthost settings (the receiver list must exist and the smarthost enabled status)
- readme amended with command examples

### CODE REVIEW

File: imageroot/actions/configure-module/validate-input.json [hunk 1]
```diff
@@ -1,8 +1,52 @@
...
+        "bantime",
+        "dyn_bantime"
+    ],
+    "properties": {
+        "helo_host": {
+            "anyOf": [
+                {
+                    "type": "string",
+                    "format": "idn-hostname",
+                    "description": "Helo host value for smtp"
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:27]:
```suggestion
                    "description": "Helo host value for SMTP"
```

File: imageroot/actions/configure-module/validate-input.json [hunk 2]
```diff
@@ -1,8 +1,52 @@
...
+                },
+                {
+                    "description": "Helo host value for smtp",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:30]:
```suggestion
                    "description": "Helo host value for SMTP",
```

File: imageroot/actions/configure-module/validate-input.json [hunk 3]
```diff
@@ -1,8 +1,52 @@
...
+        },
+        "receiver_emails": {
+            "type": "string",
+            "title": "receiver_emails",
+            "description": "List of administrator to send crowdsec alerts"
+        },
+        "bantime": {
+            "type": "string",
+            "title": "bantime",
+            "description": "ban time duration of bots (m:minute, h:hour)"
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:44]:
```suggestion
            "description": "Ban time duration of bots (m:minute, h:hour)"
```

File: imageroot/actions/configure-module/20configure [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+
+helo_host = data.get("helo_host", "")
+receiver_emails = data.get("receiver_emails", "")
+bantime =data.get("bantime", "1m")
+dyn_bantime = data.get("dyn_bantime", True)
+
+# Setup default values
+agent.set_env("HELO_HOST", helo_host)
+agent.set_env("RECEIVER_EMAILS", receiver_emails)
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/20configure:26]: I'd use an array for this parameter.
```suggestion
agent.set_env("RECEIVER_EMAILS",",".join(receiver_emails))
```

File: imageroot/actions/configure-module/validate-input.json [hunk 5]
```diff
@@ -1,8 +1,52 @@
...
-    "$id": "http://schema.nethserver.org/crowdsec/configure-module-input.json",
+    "title": "Configure crowdsec",
+    "$id": "http://nethserver.org/json-schema/task/input/crowdsec/configure-module",
     "description": "Configure crowdsec",
-    "examples": [],
-    "type": "object"
+    "examples": [
+        {
+            "helo_host": "foo.domain.com",
+            "receiver_emails": "user@domain.com,user@domain.org",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:9]: Can we use and array?
It's simpler to use and also to validate inside the UI.
```suggestion
            "receiver_emails": ["user@domain.com", "user@domain.org"],
```

File: imageroot/actions/create-module/50start-bouncer [hunk 6]
```diff
@@ -17,18 +17,17 @@ After=${MODULE_ID}.service
...
 [Service]
 Restart=always
-ExecStartPre=/usr/bin/bash -c "if ! ipset list crowdsec-blacklists > /dev/null; then firewall-cmd --permanent --new-ipset=crowdsec-blacklists --type=hash:ip --option="timeout=0" --option="maxelem=150000"; fi"
-ExecStartPre=/usr/bin/bash -c "if ! ipset list crowdsec6-blacklists > /dev/null; then firewall-cmd --permanent --new-ipset=crowdsec6-blacklists --option=family=inet6 --type=hash:ip --option="timeout=0" --option="maxelem=150000"; fi"
-ExecStartPre=firewall-cmd --reload
-ExecStartPost=firewall-cmd --direct --add-rule ipv4 filter INPUT 0 -p all -m set --match-set crowdsec-blacklists src -j DROP
-ExecStartPost=firewall-cmd --direct --add-rule ipv6 filter INPUT 0 -p all -m set --match-set crowdsec6-blacklists src -j DROP
-ExecStopPost=firewall-cmd --direct --remove-rule ipv4 filter INPUT 0  -p all -m set --match-set crowdsec-blacklists src -j DROP
-ExecStopPost=firewall-cmd --direct --remove-rule ipv6 filter INPUT 0  -p all -m set --match-set crowdsec6-blacklists src -j DROP
+ExecStartPre=/bin/bash -c "/usr/local/sbin/firewall-rules create-ipset"
```
Comment (gsanchietti) [inline imageroot/actions/create-module/50start-bouncer:20]: Why not calling it directly like:
```suggestion
ExecStartPre=/usr/local/sbin/firewall-rules create-ipset
```

File: imageroot/actions/create-module/50start-bouncer [hunk 7]
```diff
@@ -17,18 +17,17 @@ After=${MODULE_ID}.service
 
 [Service]
 Restart=always
-ExecStartPre=/usr/bin/bash -c "if ! ipset list crowdsec-blacklists > /dev/null; then firewall-cmd --permanent --new-ipset=crowdsec-blacklists --type=hash:ip --option="timeout=0" --option="maxelem=150000"; fi"
-ExecStartPre=/usr/bin/bash -c "if ! ipset list crowdsec6-blacklists > /dev/null; then firewall-cmd --permanent --new-ipset=crowdsec6-blacklists --option=family=inet6 --type=hash:ip --option="timeout=0" --option="maxelem=150000"; fi"
-ExecStartPre=firewall-cmd --reload
```
Comment (gsanchietti) [inline imageroot/actions/create-module/50start-bouncer:22]: IN the previous implementation you called `firewall-cmd --reload`, but the same command is not present inside `firewall-rules` script. Is that correct?

Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer:22]: we does it inside the script file

https://github.com/NethServer/ns8-crowdsec/pull/10/files#diff-30aeaf113aa5889f770cb8c5876e8d322d1910c16999d8aac147704f59ee19f5R16

File: imageroot/actions/create-module/50start-bouncer (hunk 6 above)
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer:20]: new attempt but this time it works....probably my VM

Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer]: done

**Review by gsanchietti — APPROVED** [review]
LGTM

File: imageroot/actions/create-module/50start-bouncer [hunk 8]
```diff
@@ -17,18 +17,17 @@ After=${MODULE_ID}.service
...
-ExecStartPost=firewall-cmd --direct --add-rule ipv4 filter INPUT 0 -p all -m set --match-set crowdsec-blacklists src -j DROP
-ExecStartPost=firewall-cmd --direct --add-rule ipv6 filter INPUT 0 -p all -m set --match-set crowdsec6-blacklists src -j DROP
-ExecStopPost=firewall-cmd --direct --remove-rule ipv4 filter INPUT 0  -p all -m set --match-set crowdsec-blacklists src -j DROP
-ExecStopPost=firewall-cmd --direct --remove-rule ipv6 filter INPUT 0  -p all -m set --match-set crowdsec6-blacklists src -j DROP
+ExecStartPre=/usr/local/sbin/firewall-rules create-ipset
+ExecStartPost=/usr/local/sbin/firewall-rules add-rule
+ExecStopPost=/usr/local/sbin/firewall-rules remove-rule
 EOF
 mkdir -vp /etc/systemd/system/crowdsec-firewall-bouncer.service.d/
 install -v -m 644 "${tmpfile}" /etc/systemd/system/crowdsec-firewall-bouncer.service.d/crowdsec-firewall-bouncer-override.conf
```
[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer]: done

File: imageroot/actions/destroy-module/60remove-firewalld-configuration [hunk 9]
```diff
@@ -11,7 +11,8 @@ set -e
 echo "Remove ipset firewall rules"
 firewall-cmd --permanent --delete-ipset=crowdsec6-blacklists
 firewall-cmd --permanent --delete-ipset=crowdsec-blacklists
-
+# remove the wrapper to add rules
+rm -f /usr/local/sbin/firewall-rules
```
[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/destroy-module/60remove-firewalld-configuration]: done

File: imageroot/actions/create-module/50start-bouncer [hunk 10]
```diff
@@ -17,18 +17,17 @@ After=${MODULE_ID}.service
...
 
 systemctl daemon-reload
 
+# create the firewall wrapper (copy it from bin/firewall-rules)
+install -v -m 0755 ../bin/firewall-rules /usr/local/sbin/firewall-rules
```
[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer]: done

File: imageroot/actions/create-module/50start-bouncer [hunk 11]
```diff
@@ -17,18 +17,14 @@ After=${MODULE_ID}.service
...
 EOF
 mkdir -vp /etc/systemd/system/crowdsec-firewall-bouncer.service.d/
-install -v -m 644 "${tmpfile}" /etc/systemd/system/crowdsec-firewall-bouncer.service.d/crowdsec-firewall-bouncer-override.conf
+install -v -m 644 "${tmpfile}" "/etc/systemd/system/crowdsec-firewall-bouncer.service.d/${MODULE_ID}-override.conf"
 
+# reload and start service
 systemctl daemon-reload
-
 systemctl enable --now crowdsec-firewall-bouncer.service
 systemctl restart "${MODULE_ID}.service"
```
[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer:30]: unfortunately there is no command to reload the container, I am thinking to remove the command we put in the systemd service, I am not user it works well

after we have started the bouncer, we need to connect the service crowdsec to the bouncer

[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer:30]: I will test it , maybe you are right

[the remark this answers is not in the document]
Comment (stephdl) [reply imageroot/actions/create-module/50start-bouncer:30]: I tested, once installed the bouncer is well registered to the crowdsec service with restarting it, you were right

```
[root@R2-pve ~]# cscli bouncers list
INFO[30-03-2023 08:56:55] Prepending yaml: '/etc/crowdsec/profiles.yaml' with '/etc/crowdsec/profiles.yaml.local' 
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 Name        IP Address   Valid   Last API pull          Type                        Version                                                    Auth Type 
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 localhost   127.0.0.1    ✔️       2023-03-30T08:56:45Z   crowdsec-firewall-bouncer   v0.0.25-el9-rpm-0a4fde8e9440927d02ce187d1716306af9a13780   api-key 
```

**Pull request merged**